aptos-move/aptos-release-builder/src/components/framework.rs

```diff
@@ -2,23 +2,31 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use anyhow::Result;
+use aptos_crypto::HashValue;
 use aptos_temppath::TempPath;
 use std::process::Command;
 
-pub fn generate_upgrade_proposals(is_testnet: bool) -> Result<Vec<(String, String)>> {
-    let package_path_list = vec![
+pub fn generate_upgrade_proposals(
+    is_testnet: bool,
+    next_execution_hash: &str,
+) -> Result<Vec<(String, String)>> {
+    let mut package_path_list = vec![
         ("0x1", "aptos-move/framework/move-stdlib"),
         ("0x1", "aptos-move/framework/aptos-stdlib"),
         ("0x1", "aptos-move/framework/aptos-framework"),
         ("0x3", "aptos-move/framework/aptos-token"),
     ];
 
-    let mut result = vec![];
+    let mut result: Vec<(String, String)> = vec![];
 
     let mut root_path = std::path::Path::new(env!("CARGO_MANIFEST_DIR")).to_path_buf();
     root_path.pop();
     root_path.pop();
 
+    if !next_execution_hash.is_empty() {
```

Comment:
Can you add a comment here to explain why reversing here would ensure the right order for multi-step proposals?

Reply:
yep will add more comments!